Create Blog Post “endless-faqs-on-whatsapp-with-generative-ai-and-ai-studio” (#2568)
Nexmo/deved-platform: content/blog/en/endless-faqs-on-whatsapp-with-generative-ai-and-ai-studio.md

---

> title: Endless FAQs on Whatsapp With Generative AI and AI Studio

Reviewer: ```suggestion
title: Endless FAQs on WhatAapp With Generative AI and AI Studio
```

  experience by building WhatsApp agents with Vonage’s low code platform; AI
  Studio.
author: benjamin-aronov
published: true
published_at: 2023-08-19T06:20:03.423Z
updated_at: 2023-08-19T06:20:03.447Z
category: tutorial
tags:
  - generative-ai
  - ai-studio
  - ai
comments: true
spotlight: false
redirect: ""
canonical: ""
outdated: false
replacement_url: ""
---
## T﻿able of Contents

1. [Introduction](#introduction)

   * [Using Generative AI to Improve Customer Experience](using-generative-ai-to-improve-customer-experience)
   * [Mission: Turning Static FAQs into an Interactive WhatsApp Agent](mission:-turning-static-faqs-into-an-interactive-whatsapp-agent)
   * [Prerequisites](#prerequisites)
2. [How to Create an AI Studio WhatsApp Agent](#how-to-create-an-ai-studio-whatsapp-agent)
3. [How to Use the AI Studio Generative AI Node](#how-to-use-the-ai-studio-generative-ai-node)
4. [How to Handle Our Generative AI Response](#how-to-handle-our-generative-ai-response)
5. [How to Use the Send Email Node in AI Studio](#how-to-use-the-send-email-node-in-ai-studio)
6. [Conclusion](#conclusion)

## Introduction

### Using Generative AI to Improve Customer Experience

> In 2023, Generative Artificial Intelligence (AI) and Large Language Models (LLM) exploded on the scene, with the promise to completely transform the world. According to [SimilarWeb](https://www.theguardian.com/technology/2023/feb/02/chatgpt-100-million-users-open-ai-fastest-growing-app), in just 2 months [ChatGPT](https://chat.openai.com/) reached 100 million users. By comparison it took TikTok 9 months and Instagram 2 years to achieve the feat.

Reviewer: ```suggestion
In 2023, Generative Artificial Intelligence (AI) and Large Language Models (LLM) exploded on the scene, with the promise to completely transform the world. According to [SimilarWeb](https://www.theguardian.com/technology/2023/feb/02/chatgpt-100-million-users-open-ai-fastest-growing-app), in just 2 months [ChatGPT](https://chat.openai.com/) reached 100 million users. By comparison, it took TikTok 9 months and Instagram 2 years to achieve the feat.
```

> But what is the significance of this booming technology? I’ve always heard that the best developers are lazy developers. I confess, I’m a lazy developer. Generative AI helps us be even lazier! Not only can we ask for refactored code or even wholly coded applications from this predictive AI. But we can outsource the other annoying parts of our applications like all the data values we want to give our users.

Reviewer: ```suggestion
But what is the significance of this booming technology? I’ve always heard that the best developers are lazy developers. I confess I’m a lazy developer. Generative AI helps us be even lazier! Not only can we ask for refactored code or even wholly coded applications from this predictive AI. But we can outsource the other annoying parts of our applications, like all the data values we want to give our users.
```

> Using Generative AI with another tool for lazy efficient developers, Vonage’s low code platform called AI Studio, you can now improve your customer experience exponentially. In this article you will learn how to replace your company’s Frequently Asked Questions (FAQ) site and turn it into a flexible, comprehensive FAQ WhatsApp agent (also known as a chatbot).

Reviewer: ```suggestion
Using Generative AI with another tool for lazy efficient developers, Vonage’s low code platform called AI Studio, you can now improve your customer experience exponentially. In this article, you will learn how to replace your company’s Frequently Asked Questions (FAQ) site and turn it into a flexible, comprehensive FAQ WhatsApp agent (also known as a chatbot).
```

---
title: Endless FAQs on Whatsapp With Generative AI and AI Studio

> description: Learn how to leverage Generative AI to enhance your customer
>   experience by building WhatsApp agents with Vonage’s low code platform; AI
>   Studio.

Reviewer: ```suggestion
description: Learn how to leverage Generative AI to enhance your customer
  experience by building WhatsApp agents with Vonage’s low code platform, AI
  Studio.
```

author: benjamin-aronov
published: true
published_at: 2023-08-19T06:20:03.423Z
updated_at: 2023-08-19T06:20:03.447Z
category: tutorial
tags:
  - generative-ai
  - ai-studio
  - ai
comments: true
spotlight: false
redirect: ""
canonical: ""
outdated: false
replacement_url: ""
---
## T﻿able of Contents

1. [Introduction](#introduction)

   * [Using Generative AI to Improve Customer Experience](using-generative-ai-to-improve-customer-experience)

>    * [Mission: Turning Static FAQs into an Interactive WhatsApp Agent](mission:-turning-static-faqs-into-an-interactive-whatsapp-agent)

Reviewer: This link is not working. Check the [preview of this blog post](https://github.com/Nexmo/deved-platform/blob/9065ef27a835a7c3f25fbf5f7e4a78c89f55fdde/content/blog/en/endless-faqs-on-whatsapp-with-generative-ai-and-ai-studio.md)

But what is the significance of this booming technology? I’ve always heard that the best developers are lazy developers. I confess, I’m a lazy developer. Generative AI helps us be even lazier! Not only can we ask for refactored code or even wholly coded applications from this predictive AI. But we can outsource the other annoying parts of our applications like all the data values we want to give our users.

Using Generative AI with another tool for lazy efficient developers, Vonage’s low code platform called AI Studio, you can now improve your customer experience exponentially. In this article you will learn how to replace your company’s Frequently Asked Questions (FAQ) site and turn it into a flexible, comprehensive FAQ WhatsApp agent (also known as a chatbot).

> > TLDR; If you don’t want to build out the Agent, you can copy (these files)\[add link to zip file] and import them to AI Studio following [these instructions](https://studio.docs.ai.vonage.com/ai-studio/agent-building-features#export-and-import-agents). You’ll still need to do the [Generative AI setup](#configuring-your-generative-ai-account-in-ai-studio).

Reviewer: Add this link before review is final.

> ![Turn Static FAQ Sites into Interactive WhatsApp Agents](/content/blog/endless-faqs-on-whatsapp-with-generative-ai-and-ai-studio/screenshot-2023-08-23-at-12.00.05.png "static-faq-into-whatsapp-agents.png")

Reviewer: In the preview, this image is broken. Not sure if it's a bug, but please re-upload this image to be on the safe side, please.

![FAQ in AI Studio without Generative AI Mockup](https://lh4.googleusercontent.com/ZI1vawCTY_blI8KlAAXiAu0d2i4vYTDPW3jDQGEnFViIBVoctYbQu1SCqQm5KlmrjNNEi2idG_wclTb_K6NTtV-4yaYU5JIGwYSptNswdBNNF4e2ox3Mlwfpy4Q4g1PNcP_iu8ISnMIfm1PKn-8DvPQ "faq-without-gai-mockup.png")

You will notice that we must create an intent not just for each action phrase that we will listen for but also for each FAQ question and answer pairing. But we are lazy builders and this would take soooo long!

Thank goodness with the Generative AI node we can simplify our agent drastically. Now our mockup looks like this:

![FAQ in AI Studio with Generative AI Mockup](https://lh3.googleusercontent.com/rtyYInT7qHdLtYioYlC5_BFh4PaJOgTr5vabpNeB0lfA2kWBL6nGJ0UfDOJWu1w4dtdQKsc5rJm5TCAS0plKqOYmQ8E049nNYPf52Tz0vKW3FG6E8LegaeOFxHS_z-StbSDTmsKIEqEE3BMyFd6p7gk "faq-with-gai-mockup.png")

Notice that now our Generative AI node will eliminate the need to define individual question/answer pairings. So in our final WhatsApp Agent who will answer FAQ questions about AI Studio, there will be a single action intent for emailing support and the Generative AI node will handle all other inputs. 

### Prerequisites

> * In order to use the Generative AI node, you will need to set up a paid account directly with OpenAI. Sign up with OpenAI using [this link](https://auth0.openai.com/u/login/identifier?state=hKFo2SBHYXZ4WkxIdWgxQWhMdTMybHNHTVJYWjJaZjRLbmlkYqFur3VuaXZlcnNhbC1sb2dpbqN0aWTZIGE5b0RUbFE2Skd4SDVGcUZoUUdoZDdfWVoxUzlqemx0o2NpZNkgRFJpdnNubTJNdTQyVDNLT3BxZHR3QjNOWXZpSFl6d0Q) and create an account.

Reviewer: ```suggestion
* In order to use the Generative AI node, you will need to set up a paid account directly with OpenAI. [Sign up with OpenAI using this link](https://auth0.openai.com/u/login/identifier?state=hKFo2SBHYXZ4WkxIdWgxQWhMdTMybHNHTVJYWjJaZjRLbmlkYqFur3VuaXZlcnNhbC1sb2dpbqN0aWTZIGE5b0RUbFE2Skd4SDVGcUZoUUdoZDdfWVoxUzlqemx0o2NpZNkgRFJpdnNubTJNdTQyVDNLT3BxZHR3QjNOWXZpSFl6d0Q) and create an account.
```

* You will need a Vonage Developer Account

<sign-up></sign-up>

### Configuring Your Generative AI Account in AI Studio

To be able to follow this tutorial you will need to make sure to have your AI Studio account up to date with the latest Generative AI integration instructions. Follow [these instructions](https://studio.docs.ai.vonage.com/whatsapp/nodes/integrations/generative-ai-coming-soon/setting-up-generative-ai-node-integration).

## How to Create an AI Studio WhatsApp Agent

To begin we’ll need to create a WhatsApp agent. You can open AI Studio from inside the Vonage [Developer Dashboard](https://dashboard.nexmo.com/) and then follow the instructions found in the [AI Studio documentation](https://studio.docs.ai.vonage.com/ai-studio/create-a-new-agent?adobe_mc=MCMID%3D58036463252966980120666334846055693945%7CMCORGID%3DA8833BC75245AF9E0A490D4D%2540AdobeOrg%7CTS%3D1692275723). There are three important options for our agent, select:

* **T﻿ype**: WhatsApp
* **Template**: S﻿tart From Scratch
* **E﻿vent**: Inbound

### How to Turn Actions Phrases into User Intents

> The first thing we need to do is build our agent until the “fork in the road”; the classification node which will decipher whether the user wants to email support (our sole action phrase) or has a question to answer.

Reviewer: ```suggestion
The first thing we need to do is build our agent until the "fork in the road", the classification node, which will decipher whether the user wants to email support (our sole action phrase) or has a question to answer.
```

Classification nodes rely on user [intents](https://studio.docs.ai.vonage.com/properties-1/intents). For each action phrase that you want to handle in your agent, you will need to create a matching intent. You can see how to [create an intent](https://studio.docs.ai.vonage.com/properties-1/intents#creating-an-intent) in the documentation.

> For our agent we need just one intent, let’s call it `EMAIL_SUPPORT`. Then let’s add some training data so the AI can understand what sort of user responses it should route to this intent. The really cool thing about AI Studio is that once you give a few user expressions, it will start to suggest more training data. The more training data you provide it, the more accurate it will sort user input. You can see some of the user expressions I provided:

Reviewer: ```suggestion
For our agent we need just one intent, let’s call it `EMAIL_SUPPORT`. Then let's add some training data so the AI can understand what sort of user responses it should route to this intent. The really cool thing about AI Studio is that once you give a few user expressions, it will start to suggest more training data. The more training data you provide it, the more accurate it will sort user input. You can see some of the user expressions I provided:
```

![Email Support Intent Example](https://lh6.googleusercontent.com/r4vwHx23KXB8DMCZqUSZzAtzJw77izMRgxQFzMIbzD7V_hgEo2EvxfeoW7CAUL3tGot3laQddbtOmPaLWymQdNTIBeeE5DFCgvK2YpQymVv3icf0ptNif8zAsUciYT00AogkTlRJpJCEDydCcCbWrkk "email-support-intent-example.png")

Make sure to save your new intent and then we can start to build out our flow!

### How to Collect User Input and Use a Classification Node

> 1. Let’s greet our user with a welcome message. Let’s use the [Send Message](https://studio.docs.ai.vonage.com/whatsapp/nodes/conversation/text-message) node. I like to attach a photo to our welcome message to use our company branding.

Reviewer: ```suggestion
1. Let's greet our users with a welcome message. Let’s use the [Send Message](https://studio.docs.ai.vonage.com/whatsapp/nodes/conversation/text-message) node. I like to attach a photo to our welcome message to use our company branding.
```

> 2. Next we’ll accept our users response with a [Collect Input](https://studio.docs.ai.vonage.com/whatsapp/nodes/conversation/collect-input) node. We’ll store the response in a [parameter](https://studio.docs.ai.vonage.com/properties-1/parameters) called `INQUIRY`. The expected input should be only text.

Reviewer: ```suggestion
2. Next we'll accept our users response with a [Collect Input](https://studio.docs.ai.vonage.com/whatsapp/nodes/conversation/collect-input) node. We’ll store the response in a [parameter](https://studio.docs.ai.vonage.com/properties-1/parameters) called `INQUIRY`. The expected input should be only text.
```

> \

Reviewer: ```suggestion

```

> Unless otherwise specified all parameters in this blog post will use the `@sys.any` entity, for more information about entities (different types of parameters)  in AI Studio please [read the documentation](https://studio.docs.ai.vonage.com/properties-1/entities).

Reviewer: ```suggestion
Unless otherwise specified, all parameters in this blog post will use the `@sys.any` entity. For more information about entities (different types of parameters) in AI Studio, please [read the documentation](https://studio.docs.ai.vonage.com/properties-1/entities).
```

3. Our third node will be a [Classification](https://studio.docs.ai.vonage.com/whatsapp/nodes/conversation/classification) node. Our Classification Parameter is `INQUIRY`. And now we can add our `EMAIL_SUPPORT` intent to the node. Your node should look like this:

![Email Support Classification Example](https://lh5.googleusercontent.com/zLoCefUccEJ35X6Dcij7Rzg4dmmEUAnMaULJsdNn5Y_CoPIPwkNAig_C4Mh4SnArPhQ9KgSlXI3l6CYQ35EyVbYtGvJ2zBOCDl_o-dUGIQdhnxKYugVN4IO_BOSBslNQ2myC6_C2r6FuUwQoeaD44qk "email-support-classification-example.png")

## How to Use the AI Studio Generative AI Node

> We’ll get back to the email support side of this agent. But first, let’s do Generative AI side. Select the Generative AI node from the Integrations section, like so: 

Reviewer: ```suggestion
We'll get back to the email support side of this agent. But first, let’s do the Generative AI side. Select the Generative AI node from the Integrations section like so: 
```

* Collect User Question

  * Using a Collect Input node ask the user for the question that was unanswerable by the agent

    * E.g. “Great! Please explain the question you have and I'll pass it along to support 💌”
  * Store it inside a new Parameter `EMAIL_INQUIRY`

Now we are ready for the Send Email node. The configuration for our node will be:

* To: support\[at]api.vonage.com

  * Make sure to hit enter so that the address is saved, it should turn into a grey block
  * You can add more than one email recipient
* Subject: New FAQ Agent Question: ﻿`$PROFILE_NAME`

  * `$PROFILE_NAME` is a [System Parameter](https://studio.docs.ai.vonage.com/properties-1/parameters#system-parameters) that allows us to access the user’s profile named they set in WhatsApp. Note: not everyone uses their real name as a profile name.
* Body:

> **Contact Info:**
>
> * Respondent Name: ﻿﻿﻿`$PROFILE_NAME`
> * Respondent Phone Number: ﻿﻿﻿`$SENDER_PHONE_NUMBER`
> * Respondent Email: ﻿﻿﻿`$USER_EMAIL`
>
> **User Submitted Question:**
>
> `$EMAIL_INQUIRY`
>
> **Conversation Time:**
>
> ﻿﻿`$CONVERSATION_START_TIME`

Your node should look like this:

> ![Email Support Node Example](/content/blog/endless-faqs-on-whatsapp-with-generative-ai-and-ai-studio/screenshot-2023-08-19-at-17.44.38.png "email-support-node-example.png")

Reviewer: In the preview, this image is broken. Not sure if it's a bug, but please re-upload this image to be on the safe side, please.

Now we can connect our Email Support node to the Collect Repeat node. This allows our user to ask more questions or end the conversation.

And that’s it! Our agent is fully built!

## Conclusion

> 👏Give yourself a round of applause! You did it! You built your first Generative AI WhatsApp agent. Now let’s test using the built-in [tester](https://studio.docs.ai.vonage.com/ai-studio/chatbot-tester).

Reviewer: ```suggestion
👏 Give yourself a round of applause! You did it! You built your first Generative AI WhatsApp agent. Now let’s test using the built-in [tester](https://studio.docs.ai.vonage.com/ai-studio/chatbot-tester).
```
